### PR TITLE
Adding ability to run ICAD tests independently of SIMD tests

### PR DESCRIPTION
**Pull request by chatton**
## Description

Run this task with 

```bash
gh workflow run "Manual E2E (Icad)" --ref "cian/issue#2015-run-icad-tests-alongside-simd-tests" -f test-entry-point=TestInterchainAccountsTestSuite -f chain-a-tag="v0.3.2" -f chain-b-tag="v0.2.2"
```
The result of this workflow can be found [here](https://github.com/cosmos/ibc-go/runs/7881282928?check_suite_focus=true)

closes: #2015

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: .github/workflows/e2e-fork.yml [hunk 1]
```diff
@@ -21,11 +21,13 @@ jobs:
           go-version: 1.18
       - id: set-matrix
         run: echo "::set-output name=matrix::$(go run cmd/build_test_matrix/main.go)"
+        env:
+          TEST_EXCLUSIONS: "TestInterchainAccountsTestSuite"
```
Comment (chatton) [inline .github/workflows/e2e-fork.yml:25]: we won't run ICA tests when there are PRs from forksin `ibc-go`

File: .github/workflows/e2e-manual-icad.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      chain-a-tag:
+        description: 'The tag to use for chain A'
+        required: true
+        type: choice
+        default: master
+        options:
+          - master
+          - v0.3.2
+          - v0.2.2
+          - v0.1.2
```
Comment (chatton) [inline .github/workflows/e2e-manual-icad.yaml:28]: these versions all include [this fix](https://github.com/cosmos/interchain-accounts-demo/pull/111) to allow for cross version testing

File: e2e/interchain_accounts_test.go [hunk 3]
```diff
@@ -14,18 +14,12 @@ import (
...
 	"github.com/cosmos/ibc-go/e2e/testvalues"
 	ibctesting "github.com/cosmos/ibc-go/v5/testing"
 )
 
 func TestInterchainAccountsTestSuite(t *testing.T) {
-	// NOTE: this is a temporary mechanism to enable this test to run alongside the simd tests.
-	// This will be removed in a follow up PR and properly parameterized in a github workflow.
-	testconfig.SetChainBinaryVersions(
-		"ghcr.io/cosmos/ibc-go-icad", "master", "icad", "ghcr.io/cosmos/ibc-go-icad", "master",
-	)
```
Comment (chatton) [inline e2e/interchain_accounts_test.go:28]: this all comes from CI now

File: e2e/testconfig/testconfig.go [hunk 4]
```diff
@@ -8,27 +8,31 @@ import (
...
-	defaultRlyTag    = "main"
+	// defaultRlyTag is the tag that will be used if no relayer tag is specified.
+	defaultRlyTag = "main"
 )
 
+func getChainImage(binary string) string {
+	if binary == "" {
+		binary = defaultBinary
+	}
+	return fmt.Sprintf("ghcr.io/cosmos/ibc-go-%s", binary)
```
Comment (colin-axner) [inline e2e/testconfig/testconfig.go:33]: Interesting. I wonder how useful this will be outside of this context.

**Review by colin-axner — APPROVED** [review]

File: e2e/testconfig/testconfig.go (hunk 4 above)
Comment (chatton) [reply e2e/testconfig/testconfig.go:33]: I don't think outside of this context there would be much use for this. The main thing is that we can always specify `CHAIN_IMAGE` to overwrite it.

Comment (colin-axner) [reply e2e/testconfig/testconfig.go:33]: Yea that's what I was thinking. I guess we won't be able to remove it until we stop supporting v4

**Review by seantking — APPROVED** [review]

**Review by damiannolan — APPROVED** [review]
❤️

**Pull request merged**